include/seqan3/alignment/matrix/alignment_trace_matrix.hpp

```diff
@@ -98,29 +99,108 @@ struct alignment_coordinate
     size_t seq2_pos;
 };
 
-/*!\brief Compute the trace from a trace matrix
+/*!\brief Compute the number of insertions, deletions and
+ * substitutions between begin coordinate and end coordinate.
  * \ingroup alignment_matrix
- * \tparam    database_t     The type of the database sequence.
- * \tparam    query_t        The type of the query sequence.
  * \tparam    trace_matrix_t The type of the trace matrix.
- * \param[in] database       The database sequence.
- * \param[in] query          The query sequence.
  * \param[in] matrix         The trace matrix.
  * \param[in] end_coordinate Where the trace in the matrix ends.
  */
+template <typename trace_matrix_t>
+//!\cond
+    requires matrix_concept<remove_cvref_t<trace_matrix_t>> &&
+             std::Same<typename remove_cvref_t<trace_matrix_t>::entry_type, trace_directions>
+//!\endcond
+inline std::tuple<size_t, size_t, size_t>
+alignment_trace_statistics(trace_matrix_t && matrix, alignment_coordinate const end_coordinate)
+{
+    using signed_size_t = std::make_signed_t<size_t>;
+    constexpr auto N = trace_directions::none;
+    constexpr auto D = trace_directions::diagonal;
+    constexpr auto L = trace_directions::left;
+    constexpr auto U = trace_directions::up;
+    size_t col = end_coordinate.seq1_pos + 1;
+    size_t row = end_coordinate.seq2_pos + 1;
+    size_t inserts = 0;
+    size_t deletes = 0;
+    size_t substitutes = 0;
+
+    while(row >= 0 && col >= 0)
```

Comment:
`row` and `col` never drop below 0 but you break when the trace is not L/U/D and one is zero.. why this check then? you could just use `while(true)`